hieroplatform/src/test/java/org/hiero/sketch/SampleCorrTest.java

```diff
@@ -0,0 +1,88 @@
+package org.hiero.sketch;
+
+import org.hiero.dataset.ParallelDataSet;
+import org.hiero.sketches.*;
+import org.hiero.table.SmallTable;
+import org.hiero.table.api.ITable;
+import org.hiero.utils.TestTables;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class SampleCorrTest {
+
+    @Test
+    public void SampleCorrTest1() {
+        int size = 40000;
+        int range = 50000;
+        int numCols = 5;
+        SmallTable data = TestTables.getCorrelatedCols(size, numCols, range);
+        //System.out.println(data.toLongString(20));
+        List<String> cn = new ArrayList<String>();
+        for (int i = 0; i < numCols; i++) {
+            cn.add("Col" + String.valueOf(i));
+        }
+        for (int i = 0; i <= 3; i++) {
+            SampleCorrelationSketch ip = new SampleCorrelationSketch(cn, Math.pow(0.5, i));
+            CorrMatrix cm = ip.create(data);
+            //System.out.printf("Sampling rate %f: ", Math.pow(0.5, i));
+            for (int j = 0; j < cn.size(); j++)
+                for (int k = 0; k < cn.size(); k++) {
+                    Assert.assertEquals(cm.getCorrelationMatrix()[j][k],
+                            cm.getCorrelationWith(cn.get(j))[k], 0.001);
+                    Assert.assertEquals(cm.getCorrelationMatrix()[j][k],
+                            cm.getCorrelation(cn.get(j), cn.get(k)), 0.001);
+                }
+            for (String s : cn)
+                Assert.assertEquals(cm.getNorm(s) * cm.getNorm(s), cm.getInnerProduct(s, s), 0.1);
+        }
+    }
+
+    @Test
+    public void SampleCorrTest2() {
+        int size = 20000;
+        int range = 5000;
+        int numCols = 4;
+        int lowdim = 100;
+        double p = 0.01;
+        ITable data = TestTables.getCorrelatedCols(size, numCols, range);
+        ParallelDataSet<ITable> all = TestTables.makeParallel(data, 100);
+        List<String> cn = new ArrayList<String>();
+        for (int i = 0; i < numCols; i++) {
+            cn.add("Col" + String.valueOf(i));
+        }
+        long start, end;
+        start = System.currentTimeMillis();
+        SampleCorrelationSketch exactIP = new SampleCorrelationSketch(cn, 1);
+        ICorrelation exactCorr = all.blockingSketch(exactIP);
+        end = System.currentTimeMillis();
+        //System.out.printf("Exact took: %d\n", end -start);
+        start = System.currentTimeMillis();
+        SampleCorrelationSketch ip = new SampleCorrelationSketch(cn, p);
+        ICorrelation cm = all.blockingSketch(ip);
+        end = System.currentTimeMillis();
+        //System.out.printf("SampleIP took: %d\n", end -start);
+        start = System.currentTimeMillis();
```

Comment:
We have some testing method for timing - although it is not really complete...
Maybe you can use that one.